torch/nn/modules/rnn.py

```diff
@@ -25,6 +26,16 @@ def __init__(self, mode, input_size, hidden_size,
         self.bidirectional = bidirectional
         num_directions = 2 if bidirectional else 1
 
+        if not isinstance(dropout, numbers.Number) or not 0 <= dropout <= 1:
```

Comment:
Why would it be something else than a number

Reply:
Some people were confused by doc and put a bool here.

Reply:
Isn’t Bool a Number too? IIRC it’s a subclass

Reply:
You are absolutely correct. I added a check for bool.